remove_wheels.sh

```diff
@@ -0,0 +1,109 @@
+#!/bin/bash
+
+# fail on undefined variables
+set -u
+# Prevent pipe errors to be silenced
+set -o pipefail
+# Exit if any command exit as non-zero
+set -e
+# enable trace mode (print what it does)
+set -x
+
+# get the anaconda token from the github secrets
+#
+# this is to prevent accidental removals
+echo "Getting anaconda token from github secrets..."
+
+ANACONDA_USER="${INPUT_ANACONDA_USER}"
+ANACONDA_TOKEN="${INPUT_ANACONDA_TOKEN}"
+N_LATEST_UPLOADS="${INPUT_N_LATEST_UPLOADS}"
+
+
+# if the ANACONDA_TOKEN is empty, exit with status -1
+# this is to prevent accidental removals
+if [ -z "${ANACONDA_TOKEN}" ]; then
+  echo "ANACONDA_TOKEN is empty, exiting..."
+  exit -1
+fi
+
+# if the N_LATEST_UPLOADS is empty, exit with status -1
+# as this should be set in by the user and it is better
+# to fail on this to signal a problem. i.e.,
+# explicit is better than implicit.
+if [ -z "${N_LATEST_UPLOADS}" ]; then
+  echo "N_LATEST_UPLOADS is empty, exiting..."
+  exit -1
+fi
+
+
+# Query the package index for packages
+#
+# TODO: should be possible to alter this, since separating the workflow
+# into two steps, one for uploading and one for cleanup, should make it
+# possible for users to manually trigger the cleanup step before/after the
+# upload step has completed in their own repos instead of us having to do it.
+#
+# TODO: raises questions on how to moderate cleanups among multiple users
+# operating on the same channel, but that might be a different issue.
+curl https://raw.githubusercontent.com/scientific-python/upload-nightly-action/main/packages-ignore-from-cleanup.txt --output packages-ignore-from-cleanup.txt
+anaconda show "${ANACONDA_USER}" &> >(grep "${ANACONDA_USER}/") | \
+    awk '{print $1}' | \
+    sed 's|.*/||g' | \
+    grep -vf packages-ignore-from-cleanup.txt > package-names.txt
+
+# Remove old uploads to save space
+# Remove all _but_ the last ${N_LATEST_UPLOADS} package versions and
+# remove all package versions older than 30 days.
+if [ -s package-names.txt ]; then
+    threshold_date="$(date +%F -d '30 days ago')"
+
+    # Remember can't quote subshell as need to split on (space separated) token
+    for package_name in $(cat package-names.txt); do
+    # TODO: this outer loop can be removed when ready since there will be
+    # just one package to remove when the action is triggered manually from
+    # a user's (different) repo.
+
+        echo -e "\n# package: ${package_name}"
```

Comment:
This TODO item would also be resolved if we have a way forward with the questions above, since, if multiple packages are removed at once, this loop can stay; if we were to restrict the action to removing just one wheel at a time (which I don't think we should), then we could remove this. I think an action of the form:

```yaml
- name: Remove old wheels
  uses: scientific-python/upload-nightly-action/remove-wheels@cantknowhashyet # 0.6.0
  with:
    n_latest_uploads: ${{ env.N_LATEST_UPLOADS }}
    anaconda_nightly_upload_organization: "your-organization"
    anaconda_nightly_token: ${{secrets.ANACONDA_TOKEN}}
    packages_to_remove: "mypackage1,mypackage2,mypackage3" # or just "*"
    # I could have suggested "all" here, but that breaks in the
    # case where "all" is also the name of a package that has
    # been uploaded (possible, albeit quite unlikely)
```

is better than specifying the step and authenticating multiple times.

Reply:
`packages: ["a", "b", "c"]` hopefully!

Reply:
Yes, a list of strings sounds elegant! :)